dynawo/sources/Solvers/SolverSIM/DYNSolverSIM.cpp

```diff
@@ -494,10 +495,14 @@ SolverSIM::solve() {
       // Algebraic mode change
       modeChangeType_t modeChangeType = model_->getModeChangeType();
       if (modeChangeType == ALGEBRAIC_MODE || modeChangeType == ALGEBRAIC_J_UPDATE_MODE) {
+        skipNextNR_ = false;
         return (ROOT_ALG);
       } else if (modelChange) {  // Z change
+        skipNextNR_ = false;
         return (ROOT);
       } else {  // Root change without any z or mode change
+        if (modeChangeType == DIFFERENTIAL_MODE)
+          skipNextNR_ = false;
```

Comment:
We need to be sure of the initial value of skipNextNR_ here, but I don't see it being reset anywhere?

Reply:
I believe we don't really care about the initial value as the idea is to force to do the next Newton-Raphson resolution as soon as something has changed in the system, whatever is the previous status.

Of course, in some cases, it would already have been false for the current calculation but it doesn't have any impact to reset it, except if I forget something (no real impact on performances, isn't it?).